### PR TITLE
Move to pytest test suite

### PR DESCRIPTION
**Pull request by phracek**

### CODE REVIEW

**Comment by zmiklank** [comment]
Hello,

would it be possible to change "test" for "tests" in `setup.py` in order to install cwt package without tests/ directory?:

```
diff --git a/setup.py b/setup.py
index 89b6c41..64bfbf5 100755
--- a/setup.py
+++ b/setup.py
@@ -80,7 +80,7 @@ setup(
     author_email='pkubat@redhat.com',
     url='https://github.com/sclorg/container-workflow-tool',
     license='MIT',
-    packages=find_packages(exclude=['man', 'test']),
+    packages=find_packages(exclude=['man', 'tests']),
     include_package_data=True,
     data_files=data_files.items(),
     scripts=[],
```

**Comment by phracek** [comment]
> Hello,
> 
> would it be possible to change "test" for "tests" in `setup.py` in order to install cwt package without tests/ directory?:
> 
> ```
> diff --git a/setup.py b/setup.py
> index 89b6c41..64bfbf5 100755
> --- a/setup.py
> +++ b/setup.py
> @@ -80,7 +80,7 @@ setup(
>      author_email='pkubat@redhat.com',
>      url='https://github.com/sclorg/container-workflow-tool',
>      license='MIT',
> -    packages=find_packages(exclude=['man', 'test']),
> +    packages=find_packages(exclude=['man', 'tests']),
>      include_package_data=True,
>      data_files=data_files.items(),
>      scripts=[],
> ```

Thanks for catching this bug, I will fix it by the next commit.

**Comment by phracek** [comment]
Tests are passing. Can we merge it?

File: tests/spellbook.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+"""
+A book with our finest spells
```
Comment (pkubatrh) [inline tests/spellbook.py:25]: Can we get a comment here describing what this file should be used for?

File: tests/test_brew.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+from container_workflow_tool.cli import ImageRebuilder
+
+
+class TestBrew(object):
+    def setup_method(self):
+        self.component = 's2i-base'
+        self.ir = ImageRebuilder('Testing')
+
+        self.ir.set_config('default.yaml', release="rawhide")
+        # Partner BZ testing
```
Comment (pkubatrh) [inline tests/test_brew.py:34]: Wondering if this and other comments in other files are supposed to be there. Afaik there are no BZ features in cwt.

**Comment by pkubatrh** [comment]
Just a few nitpicks from me

**Comment by phracek** [comment]
[test]

**Pull request merged**